client/modules/User/components/SignupForm.jsx

```diff
@@ -43,16 +35,12 @@ function SignupForm() {
   return (
     <Form
       fields={['username', 'email', 'password', 'confirmPassword']}
-      validate={validateSignup}
+      validate={(values) => SignupFormInput.safeParse(values)}
       onSubmit={onSubmit}
     >
       {({ handleSubmit, pristine, submitting, invalid }) => (
         <form className="form" onSubmit={handleSubmit}>
-          <Field
-            name="username"
-            validate={validateUsername}
-            validateFields={[]}
-          >
+          <Field name="username" validateFields={[]}>
```

Comment:
The signup form has an asynchronous check where we ping the API to make sure that the username and email address are not already in use.

We definitely don’t want to lose that check.  We would want to either migrate it to Zod, or run it as an independent function on those form fields.

I think that keeping it independent might make sense because as it is right now we are pinging to API more than is necessary.  So it’s been on my list of low-priority things that I want to refactor.

That is, unless Zod has some sort of utility that allows us to call the API only when the username has changed.  That would be great if such a thing exists.  Otherwise I need to write one.